### PR TITLE
refactor: add default true type

### LINKED ISSUE #970: Config bools that default to true require more work
In #954, we discovered that booleans that default to true were being overridden when applying default values. When boolean values are processed by the defaults library, there is not way to distinguish between a `false` that was intentionally set and a zero boolean value. This results in the default boolean value being used (`true` in this case) making it not possible for a boolean that defaults to `true` to be set to `false`. 

We fixed this by setting a pointer to a boolean value so that there are three possible states for a boolean: `true`, `false` and `nil`. If we send a `nil` boolean to the defaults library it is properly interpreted as unset and can be updated to `false`. 

There is some technical debt leftover from this fix: 

- [x] add a getter function for `HoneycombLoggerConfig.SamplerEnabled` 
- [ ] add a `default-true` type in order to avoid hardcoding true values in getter functions for related fields. 
- [ ] use something (reflect?) to do struct tag lookups to find default values in the getter functions instead of declaring defaults in multiple places

### PR DESCRIPTION
**Pull request by fchikwekwe**
## Which problem is this PR solving?

- Closes #970 

## Short description of the changes

- adds a `DefaultTrue` type to use when specifying values with a default of boolean `true` this improves some getter functions by abstracting away the `*bool` behavior and allows boolean `true` be a true default

### CODE REVIEW

File: config/validate.go [hunk 1]
```diff
@@ -184,6 +184,15 @@ func validateDatatype(k string, v any, typ string) string {
 		if u.Scheme != "http" && u.Scheme != "https" {
 			return fmt.Sprintf("field %s (%v) must use an http or https scheme", k, v)
 		}
+	case "defaulttrue":
+		if !isString(v) {
+			return fmt.Sprintf("field %s must be of type DefaultTrue", k)
```
Comment (kentquirk) [inline config/validate.go:189]: This error message wouldn't be generated *unless* the field was of type DefaultTrue. This validation is checking the value portion of the configuration, so you actually want to check `if _, ok := v.(bool)` to make sure the user is specifying either true or false for the value.

I'm not sure you even need the UnmarshalText part below. But to confirm, you should try a couple of different values in a configuration and run the validator to make sure it does the right thing.

Comment (fchikwekwe) [reply config/validate.go]: This value comes in as a string, so I used `strconv.ParseBool` instead.

Comment (fchikwekwe) [reply config/validate.go]: According the docs, it does not seem that 1/0 are supported as bools yaml. I'll fix the validation to check for "true"/"false", "t"/"f".

Comment (fchikwekwe) [reply config/validate.go]: Its a little ugly, but I've updated it.

File: config/validate.go [hunk 2]
```diff
@@ -184,6 +184,17 @@ func validateDatatype(k string, v any, typ string) string {
 		if u.Scheme != "http" && u.Scheme != "https" {
 			return fmt.Sprintf("field %s (%v) must use an http or https scheme", k, v)
 		}
+	case "defaulttrue":
+		switch val := v.(type) {
+		case bool:
+			return ""
+		case string:
+			if val != "t" && val != "f" && val != "true" && val != "false" {
```
Comment (kentquirk) [inline config/validate.go:192]: Please use `strings.ToLower(val)`. Another way to do this if you prefer would be something like:
```
switch strings.ToLower(val) {
case "t", "true", "f", "false":
default:
    // error
```

Comment (fchikwekwe) [reply config/validate.go:192]: Sure, the second switch statement looks better to me.

**Review by kentquirk — APPROVED** [review]
Looks good!

**Pull request merged**